### PR TITLE
CLN,TYP: Use subsitutions in info docstrings

### PR DESCRIPTION
**Pull request by MarcoGorelli**
Another precursor to #31796, which should make it easier to review/merge

----

screenshots of docs

image
image
image
image

### CODE REVIEW

**Review by simonjayhawkins — APPROVED** [review]
Thanks @MarcoGorelli lgtm

File: pandas/core/frame.py [hunk 1]
```diff
@@ -2326,9 +2326,111 @@ def to_html(
...
+    0   column_1  1000000 non-null  object
+    1   column_2  1000000 non-null  object
+    2   column_3  1000000 non-null  object
+dtypes: object(3)
+memory usage: 188.8 MB""",
+        see_also_sub="""
+DataFrame.describe: Generate descriptive statistics of DataFrame
+    columns.
+DataFrame.memory_usage: Memory usage of DataFrame columns.""",
+    )
```
Comment (jbrockmendel) [inline pandas/core/frame.py:2425]: can you indent these

Comment (jbrockmendel) [reply pandas/core/frame.py:2425]: or keep the text adjacent to the implementation

Comment (MarcoGorelli) [reply pandas/core/frame.py:2433]: @jbrockmendel that's a good suggestion - have indented

**Comment by WillAyd** [comment]
Lgtm - @jbrockmendel merge if happy

**Comment by jbrockmendel** [comment]
Thanks @MarcoGorelli

**Pull request merged**